src/main/java/com/amazonaws/encryptionsdk/internal/DecryptionHandler.java

```diff
@@ -536,9 +539,18 @@ public CiphertextHeaders getHeaders() {
 
     @Override
     public List<K> getMasterKeys() {
+        if(dataKey_.getMasterKey() == null) {
```

Comment:
When a keyring is used, `dataKey_.getMasterKey()` will return null, so I wanted the returned list to be an empty list, rather than a list of one null element. It looks like I forgot to deprecate this method though (and a couple others in this class), I'll fix that